src/pkg/generators/host_generator.go

```diff
@@ -32,101 +33,140 @@ import (
 )
 
 type HostsSimulator struct {
-	hosts                 []devops.Host
-	appendLabels []*prompb.Label
+	hosts                 map[int][]*devops.Host
+	scrapeIntervalSeconds int
+	appendLabels          []*prompb.Label
 }
 
 type HostsSimulatorOptions struct {
 	Labels map[string]string
 }
 
 func NewHostsSimulator(
-	hostCount int, 
+	hostCount, scrapeIntervalSeconds int,
 	start time.Time,
 	opts HostsSimulatorOptions,
 ) *HostsSimulator {
-	var hosts []devops.Host
+	hosts := make(map[int][]*devops.Host, scrapeIntervalSeconds)
```

Comment:
Hm, I still don't see why there needs to be a map[..] to an array of hosts.

Can't we just remove some hosts and add them each time based on the percent of new series?  I don't think we need to use different hosts each tick, we just need to TickAll(..) then remove/add some.